### ISSUE TITLE
[Bug]: App fails when OpenLIT is used

### ISSUE THREAD

**Issue by gevial** [openlit/openlit#297]
### 🐛 What's Going Wrong?
I have a small LangChain-based app that works. When I add `openlit.init()`, it fails to start with the following exceptions:

```
Failed to instrument transformers: No module named 'transformers.pipelines'; 'transformers' is not a package
...
  File "/Users/User/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/langchain_community/embeddings/bedrock.py", line 150, in _embedding_func
    raise ValueError(f"Error raised by inference endpoint: {e}")
ValueError: Error raised by inference endpoint: 'SSO' object has no attribute 'invoke_model'
```
If I remove `transformers` package it complains about the absense:
```
DependencyConflict: requested: "transformers >= 4.39.3" but found: "None"
```

### 🕵️ Steps to Reproduce
I have an app that makes use of the following:
- langchain 0.2.5
- chromadb 0.5.3
- Amazon Bedrock with Claude 3 and Titan v2 Embeddings

Here is the code with logic stripped for brevity:
```
import openlit

from langchain_community.embeddings.bedrock import BedrockEmbeddings
from langchain_community.vectorstores.chroma import Chroma
from langchain_aws import ChatBedrock
from langchain_core.output_parsers import StrOutputParser
from langchain_core.prompts import ChatPromptTemplate

openlit.init()

text_for_search = ""

be = BedrockEmbeddings(model_id='amazon.titan-embed-text-v2:0')
ch = Chroma(persist_directory='./chroma', embedding_function=be)
sres = ch.similarity_search_with_score(query=text_for_search)

prompt = """prompt {input} {search_result}"""

llm = ChatBedrock(model_id="anthropic.claude-3-opus-20240229-v1:0")
prompt = ChatPromptTemplate.from_template(prompt)
output_parser = StrOutputParser()
chain = prompt | llm | output_parser

r = chain.invoke({"input": "some input", "search_result": sres[0][0].page_content})
```

When I comment out `openlit.init()`, it works.

### 🎯 What Did You Expect?
The app gets instrumented with traces.

### 📸 Any Screenshots?
>  Pictures can say a thousand words and can be super helpful!

### 💻 Your Setup
- OpenLIT Version: 1.14.1
- Deployment Method: Docker

### 📝 Additional Notes
>  Got more to say? Tell us here.

### 👐 Want to Help Make It Happen?
- [ ] Yes, I'd like to volunteer and help out with this!

**Comment by AmanAgarwal041** [comment]
Thanks for letting us know. We will fix it soon @gevial .

**Comment by amanagarwal042** [comment]
Hey @gevial Whats the SDK version you are using?

A quick fix would be to ignore the HuggingFace instrumentation

```
openlit.init(disabled_instrumentors=["transformers"])
```

Also I do see
ValueError: Error raised by inference endpoint: 'SSO' object has no attribute 'invoke_model'
which might be the source of app failure as failed instrumentation should ideally not break the application code, It would just log that it failed

**Comment by gevial** [comment]
Hi @patcher9 
It's version 1.14.1

I tried the fix you suggested but unfortunately the app fails in the same way.
What you're saying about instrumentation makes total sense, but the fact is the app works just fine until I add a single line of `openlit.init()` (with the import statement of course).

**Comment by amanagarwal042** [comment]
qq do you also have transformers installed as a library? do you know the version that you have installed?

**Comment by gevial** [comment]
@patcher9 I had had to add it (4.42.3) because otherwise the instrumentation failed with 

`DependencyConflict: requested: "transformers >= 4.39.3" but found: "None"`

But then I changed `openlit.init()` to `openlit.init(disabled_instrumentors=["transformers"])`, and removed transformers. I don't have `DependencyConflict` anymore, but the exception is still the same.

**Comment by amanagarwal042** [comment]
Closing this issues as fixed in OpenLIT sdk version `1.14.2`

If its still relevant, Feel free to reopen

**Comment by gevial** [comment]
Hi @patcher9 

Unfortunately the error persists. 
Here is the full trace. I suspect that may be connected to the fact I'm authenticating to AWS using SSO.

```
{
    "name": "bedrock.invoke_model",
    "context": {
        "trace_id": "0xccb8c145f0cbb83b6a9fb507f2f7298f",
        "span_id": "0x8929caf4aa31c01f",
        "trace_state": "[]"
    },
    "kind": "SpanKind.CLIENT",
    "parent_id": null,
    "start_time": "2024-07-01T16:58:04.064154Z",
    "end_time": "2024-07-01T16:58:04.098340Z",
    "status": {
        "status_code": "ERROR",
        "description": "AttributeError: 'SSO' object has no attribute 'invoke_model'"
    },
    "attributes": {},
    "events": [
        {
            "name": "exception",
            "timestamp": "2024-07-01T16:58:04.098168Z",
            "attributes": {
                "exception.type": "AttributeError",
                "exception.message": "'SSO' object has no attribute 'invoke_model'",
                "exception.stacktrace": "Traceback (most recent call last):\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/opentelemetry/trace/__init__.py\", line 570, in use_span\n    yield span\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/opentelemetry/sdk/trace/__init__.py\", line 1071, in start_as_current_span\n    yield span\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/openlit/instrumentation/bedrock/bedrock.py\", line 375, in add_instrumentation\n    response = original_method(*method_args, **method_kwargs)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/client.py\", line 565, in _api_call\n    return self._make_api_call(operation_name, kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/client.py\", line 1001, in _make_api_call\n    http, parsed_response = self._make_request(\n                            ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/client.py\", line 1027, in _make_request\n    return self._endpoint.make_request(operation_model, request_dict)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/endpoint.py\", line 119, in make_request\n    return self._send_request(request_dict, operation_model)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/endpoint.py\", line 198, in _send_request\n    request = self.create_request(request_dict, operation_model)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/endpoint.py\", line 134, in create_request\n    self._event_emitter.emit(\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/hooks.py\", line 412, in emit\n    return self._emitter.emit(aliased_event_name, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/hooks.py\", line 256, in emit\n    return self._emit(event_name, kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/hooks.py\", line 239, in _emit\n    response = handler(**kwargs)\n               ^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/signers.py\", line 105, in handler\n    return self.sign(operation_name, request)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/signers.py\", line 190, in sign\n    auth = self.get_auth_instance(**kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/signers.py\", line 310, in get_auth_instance\n    frozen_credentials = credentials.get_frozen_credentials()\n                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/credentials.py\", line 634, in get_frozen_credentials\n    self._refresh()\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/credentials.py\", line 522, in _refresh\n    self._protected_refresh(is_mandatory=is_mandatory_refresh)\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/credentials.py\", line 538, in _protected_refresh\n    metadata = self._refresh_using()\n               ^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/credentials.py\", line 685, in fetch_credentials\n    return self._get_cached_credentials()\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/credentials.py\", line 695, in _get_cached_credentials\n    response = self._get_credentials()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/credentials.py\", line 2147, in _get_credentials\n    client = self._client_creator('sso', config=config)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/session.py\", line 997, in create_client\n    client = client_creator.create_client(\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/openlit/instrumentation/bedrock/bedrock.py\", line 430, in wrapper\n    original_invoke_model = client.invoke_model\n                            ^^^^^^^^^^^^^^^^^^^\n  File \"/Users/Gennadii_Aleksandrov/Library/Caches/pypoetry/virtualenvs/librefactor-k7sXJnK--py3.12/lib/python3.12/site-packages/botocore/client.py\", line 918, in __getattr__\n    raise AttributeError(\nAttributeError: 'SSO' object has no attribute 'invoke_model'\n",
                "exception.escaped": "False"
            }
        }
    ],
    "links": [],
    "resource": {
        "attributes": {
            "service.name": "default",
            "deployment.environment": "default",
            "telemetry.sdk.name": "openlit"
        },
        "schema_url": ""
    }
}
```

**Comment by amanagarwal042** [comment]
I hope not, I am trying it on my account. 

Reopened the issue and will keep you updated with findings

**Comment by amanagarwal042** [comment]
```python
import openlit

from langchain_aws import ChatBedrock
from langchain_core.messages import HumanMessage

openlit.init()

import boto3

client = boto3.client("bedrock-runtime", aws_access_key_id="xxxx",
    aws_secret_access_key="xxxx", region_name="xxxx")

chat = ChatBedrock(
  client=client,
  model_id="amazon.titan-text-lite-v1",
)

messages = [
    HumanMessage(
        content="Translate this sentence from English to French. I love programming."
    )
]
chat.invoke(messages)
```

> I suspect that may be connected to the fact I'm authenticating to AWS using SSO.


So I directly passed my aws creds to the boto3 client and seems to work for me. cause looking the the error in the span, it seems to be related the auth itself

<details>
<summary>OpenTelemetry Trace for the LLM/Bedrock call</summary>

```json
{
    "name": "bedrock.invoke_model",
    "context": {
        "trace_id": "0x269c7b4f698262a753ee32ece56ea6bb",
        "span_id": "0xb61fd20ce01ae031",
        "trace_state": "[]"
    },
    "kind": "SpanKind.CLIENT",
    "parent_id": null,
    "start_time": "2024-07-03T18:17:06.739839Z",
    "end_time": "2024-07-03T18:17:07.973304Z",
    "status": {
        "status_code": "OK"
    },
    "attributes": {
        "gen_ai.operation.name": "chat",
        "telemetry.sdk.name": "openlit",
        "gen_ai.system": "bedrock",
        "gen_ai.endpoint": "bedrock.invoke_model",
        "gen_ai.environment": "default",
        "gen_ai.application_name": "default",
        "gen_ai.request.model": "amazon.titan-text-lite-v1",
        "gen_ai.usage.prompt_tokens": 18,
        "gen_ai.usage.completion_tokens": 16,
        "gen_ai.usage.total_tokens": 34,
        "gen_ai.response.finish_reason": "FINISH",
        "gen_ai.usage.cost": 1.1799999999999999e-05,
        "gen_ai.prompt": "\n\nUser: Translate this sentence from English to French. I love programming.\n\nBot:",
        "gen_ai.completion": " Here is the French translation: Je suis amoureux de la programmation."
    },
    "events": [],
    "links": [],
    "resource": {
        "attributes": {
            "service.name": "default",
            "deployment.environment": "default",
            "telemetry.sdk.name": "openlit"
        },
        "schema_url": ""
    }
}
```
</details>

**Comment by gevial** [comment]
Thanks for trying this out @patcher99 
That's what I thought given the error message. The auth itself works just fine, I do `aws sso login` and `boto3` uses the configured creds just fine. Unless I call `openlit.init()`, then it crashes.

**Comment by amanagarwal042** [comment]
Yeah sorry for the late reply! Cant seem to figure out whats failing as the setup seems to working for me

**Comment by gevial** [comment]
Hi @patcher9 
I wanted to check without SSO, and got a new error. 
I'm now using DynamoDB in my app, and uncommenting `openlit.init()` now raises

```
'DynamoDB' object has no attribute 'invoke_model'
```
So it has something to do with how OpenLIT interacts with boto3.

Here is my code:
```
import openlit
import analyser
import boto3

from langchain_community.embeddings.bedrock import BedrockEmbeddings
from langchain_community.vectorstores.chroma import Chroma
# from langchain_core.documents import Document
from langchain_aws import ChatBedrock
from langchain_core.output_parsers import StrOutputParser
from langchain_core.prompts import ChatPromptTemplate

openlit.init(disabled_instrumentors=["transformers"])
# openlit.init()

client = '''<skipped for brevity>'''

orig_method_sig = analyser.MethodSignature("org.elasticsearch.client.RestClient.performRequestAsync", ("String", "String", "Map<String, String>", "HttpEntity", "ResponseListener"))

st = analyser.DynamoDbSymbolTable("libupgrade_test", boto3.resource("dynamodb"), "org.elasticsearch.client:elasticsearch-rest-client:5.6.0")
```

I got exception on the line where I'm using `boto3.resource("dynamodb")`:

```
Exception has occurred: AttributeError
'DynamoDB' object has no attribute 'invoke_model'
  File "/Users/Gennadii_Aleksandrov/code/lib_upgrade_prototype/libupgrade/libupgrade/elastic.py", line 42, in <module>
    st = analyser.DynamoDbSymbolTable("libupgrade_test", boto3.resource("dynamodb"), "org.elasticsearch.client:elasticsearch-rest-client:5.6.0")
                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'DynamoDB' object has no attribute 'invoke_model'
```

When I remove `openlit.init()` the code works.

**Comment by amanagarwal042** [comment]
Yeah I think I know whats happening in this issue.

```
client = '''<skipped for brevity>'''
```

^ Can you share like a very rough snippet of this part, I wanna see how this client is created and how its different from creating a bedrock client

So we wrap 
```
botocore.client
ClientCreator.create_client
``` 

function from the boto3 library but that seems to be used for other uses cases aswell. So Ill make it more bedrock specific and hopefully that should solve the issue. Ill try to get the fix in by tommorow

**Comment by gevial** [comment]
sorry for confusion @patcher9 , 
```
client = '''<skipped for brevity>'''
```
is actually just a string literal containing some Java code I'm trying to transform with LLM.

I'm not constructing boto client there.

**Comment by amanagarwal042** [comment]
Working on the PR here
https://github.com/openlit/openlit/pull/350

I figured also that the `invoke_model` is deprecated in favor of the converse API so I am also removing monitoring/wrapping for all LLM calls made via `invoke_model` and instead adding it for `converse` as thats the recommended way

**Comment by amanagarwal042** [comment]
Just released the Python SDK version `1.1.6.2` that has the fix for you issue and replacement of `invoke_model` with `converse` function. Closing this issue as the issue should be fixed but incase if its still relevant, Feel free to open it up again (Or create a new one)

**Comment by gevial** [comment]
Hi @patcher9, many thanks for looking into it. It doesn't fail now, the app works and I can see some ChromaDB-related traces in the console.

However, I can see nothing in the OpenLIT UI that I run with Docker locally. 
Not sure it is related to this bug though.
